### PR TITLE
Refactoring - Update class setAccess and getAccess

### PR DESCRIPTION
**Pull request by akeeste**
This PR addresses some refactoring items, namely updates the ``setAccess`` and ``getAccess`` of various class properties based on how they are used. Some ``setX()`` methods are added so that properties can be private-public while still set in ``initializeWecSim``

- [x] make ``number`` properties private-public. 
- [x] Add ``setNumber()`` methods so that ``number`` properties can still be set in ``initializeWecSim``
- [x] Remove ``mooring.orientation`` as it is not used anywhere. Note, this is not used in the same way as the other classes
- [x] move ``waves.type`` to private-public. ``typeNum`` is set privately and only in the class constructor so ``type`` must be set through the constructor as well
- [x] ``body.geometry, body.h5File`` to private-public and update the docstring so that the API pulls the full description
- [x] ``body.dofCoupled, body.total`` to private-public and defined in a ``body.setup()`` method
- [x] remove ``body.hydroTotal``, not used
- [x] ``body.massCalcMethod`` to private-public and removed from ``initializeWecSim`` line 149. This should not be called there
- [x] ``simu.cicTime, simu.cicLength, simu.caseFile, simu.date, simu.gitCommit, simu.maxIt, simu.outputDir, simu.wsVersion`` to private-public, not used outside of the simulationClass
- [x] remove ``simu.inputFile, simu.logFile``, not used

### CODE REVIEW

**Comment by kmruehl** [comment]
@akeeste can you merge this into my fork so that it's included in PR into https://github.com/WEC-Sim/WEC-Sim/pull/832? Thank you

File: source/functions/initializeWecSim.m [hunk 1]
```diff
@@ -170,19 +168,13 @@
         body(ii).loadHydroData(tmp_hydroData);
         clear tmp_hydroData
     end
-    body(ii).total = simu.numHydroBodies;
-    if simu.b2b==1
-        body(ii).dofCoupled = zeros(6*body(ii).total,1);
-    else
-        body(ii).dofCoupled = zeros(6,1);
-    end
```
Comment (kmruehl) [inline source/functions/initializeWecSim.m:178]: @akeeste where was this moved?

File: source/objects/bodyClass.m [hunk 2]
```diff
@@ -440,8 +433,21 @@ function plotStl(obj)
...
+
+        function setup(obj, numHydroBodies, b2b)
+            % Method to define the bodies dofCoupled parameters
+            obj.total = numHydroBodies;
+            if b2b==1
+                obj.dofCoupled = zeros(6*numHydroBodies,1);
+            else
+                obj.dofCoupled = zeros(6,1);
+            end
+        end
```
Comment (kmruehl) [inline source/objects/bodyClass.m:450]: this method needs a clearer name, e.g. `setDOF`

File: source/objects/mooringClass.m [hunk 3]
```diff
@@ -42,10 +42,9 @@
         name                    = 'NOT DEFINED'                            % (`string`) Name of the mooring. Default = ``'NOT DEFINED'``
     end
 
-    properties (SetAccess = 'public', GetAccess = 'public') %internal
+    properties (SetAccess = 'private', GetAccess = 'public') %internal
         moorDyn                 = 0                                        % (`integer`) Flag to indicate a MoorDyn block, 0 or 1. Default = ``0``
         number                  = []                                       % (`integer`) Mooring number. Default = ``'NOT DEFINED'``        
-        orientation             = []                                            % (`float 1 x 6`) Initial 6DOF location. Default = ``[0 0 0 0 0 0]``        
     end
```
Comment (kmruehl) [inline source/objects/mooringClass.m:48]: are you sure this is unused?

**Review by kmruehl — COMMENTED** [review]
Thanks @akeeste, this looks good, but please put it into my fork so we have the last refactoring revisions in one place.

File: source/objects/mooringClass.m (hunk 3 above)
Comment (akeeste) [reply source/objects/mooringClass.m:48]: Yes this is never called in the mooring library or anywhere else in the mooring class. We probably should allow users to define the mooring orientation, but right now it's not possible. I thought to leave this for a larger discussion of how we define initial positions

File: source/functions/initializeWecSim.m [hunk 4]
```diff
@@ -122,8 +122,7 @@
 if exist('mooring','var') == 1
     simu.numMoorings = length(mooring(1,:));
     for ii = 1:simu.numMoorings
-        mooring(ii).number = ii;
-        mooring(ii).setLoc;
+        mooring(ii).setNumber(ii);
```
Comment (kmruehl) [inline source/functions/initializeWecSim.m:125]: what about `setLoc`, are we sure it's not used?

File: source/functions/initializeWecSim.m [hunk 5]
```diff
@@ -135,25 +134,24 @@
...
-    else
-        body(ii).massCalcMethod = 'user';
+        dragBodLogic(ii) = 1;
     end
 end
 simu.numHydroBodies = numHydroBodies; clear numHydroBodies
 simu.numDragBodies = numDragBodies; clear numDragBodies
 for ii = 1:simu.numHydroBodies
-    body(ii).checkinputs();
-    %Determine if hydro data needs to be reloaded from h5 file, or if hydroData
```
Comment (kmruehl) [inline source/functions/initializeWecSim.m:156]: where is `body.checkininputs()` called?

File: source/functions/initializeWecSim.m [hunk 6]
```diff
@@ -135,25 +134,24 @@
...
+    body(ii).setup(simu.numHydroBodies,simu.b2b);
+
```
Comment (kmruehl) [inline source/functions/initializeWecSim.m:153]: change method name

File: source/objects/bodyClass.m [hunk 7]
```diff
@@ -70,48 +65,46 @@
...
+            'vertex',           [], ...                     % 
+            'face',             [], ...                     % 
+            'norm',             [], ...                     % 
+            'area',             [], ...                     % 
+            'center',           [])                         % (`structure`) Defines each body's mesh. `numFace` (`integer`) Number of faces, `numVertex` (`integer`) Number of vertices, `vertex` (`numVertex x 3 float matrix`) List of vertices, `face` (`numFace x 3 float matrix`) List of faces, `norm` (`numFace x 3 float matrix`) List of normal vectors, `area` (`numFace x 1 float matrix`) List of cell areas, `center` (`numFace x 3 float matrix`) List of cell centers. Default = [].
+        massCalcMethod      = []                            % (`string`) Method used to obtain mass, options: ``'user'``, ``'fixed'``, ``'equilibrium'``
+        number              = []                            % (`integer`) Body number, must be the same as the BEM body number.
+        total               = []                            % (`integer`) Total number of hydro bodies         
     end
```
Comment (kmruehl) [inline source/objects/bodyClass.m:90]: Keep the the H5, internal and STL breakdown please

**Review by kmruehl — COMMENTED** [review]
See comments

**Comment by kmruehl** [comment]
@akeeste it looks like there are some conflicts with dev now, please resolve. I'm also going to resolve them on my branch

**Comment by kmruehl** [comment]
@akeeste tests are failing due to `waves.checkinputs();`, the error says this method doesn't exist. Any ideas?

```matlab
 Error occurred while setting up or tearing down regressionTest.
  As a result, all regressionTest tests failed and did not run to completion.
      ---------
      Error ID:
      ---------
      'MATLAB:noSuchMethodOrField'
      --------------
      Error Details:
      --------------
      Unrecognized method, property, or field 'checkinputs' for class 'waveClass'.
      
      Error in initializeWecSim (line 93)
      waves.checkinputs();
      
      Error in run (line 91)
      evalin('caller', strcat(script, ';'));
      
      Error in wecSim (line 44)
      run('initializeWecSim');
      
      Error in runLoadRegular (line 25)
      wecSim;                     % Run Simulation
```

**Comment by kmruehl** [comment]
@akeeste I think it's a case issue, `checkInputs()` noy `checkinputs()`

**Review by kmruehl — APPROVED** [review]
@akeeste I'm going to go ahead and merge this into dev

**Pull request merged**